packages/giselle-engine/src/core/generations/generate-text.ts

```diff
@@ -78,6 +85,23 @@ export async function generateText(args: {
 		startedAt: Date.now(),
 	} satisfies RunningGeneration;
 
+	const trace = langfuse.trace({
+		userId: String(args.telemetry?.metadata?.userId),
+		name: "ai.streamText",
+		metadata: args.telemetry?.metadata,
+	});
+	const span = trace.span({
+		name: "ai.streamText",
+		startTime: new Date(runningGeneration.queuedAt),
+	});
+	const generation = span.generation({
+		name: "ai.streamText.doStream",
+		model: operationNode.content.llm.id,
+		modelParameters: operationNode.content.llm.configurations,
+		startTime: new Date(runningGeneration.createdAt),
+		completionStartTime: new Date(runningGeneration.startedAt),
+	});
+
```

Comment:
Why do you need to directly operate the Langfuse instance? I would like to know the reason why it cannot be done within the framework of AI SDK Telemetry.
https://ai-sdk.dev/docs/ai-sdk-core/telemetry

Also, if it cannot be done within the framework of AI SDK Telemetry, is it not possible to create a Custom Tracer to address this?
https://ai-sdk.dev/docs/ai-sdk-core/telemetry#custom-tracer

Reply:
> Why do you need to directly operate the Langfuse instance?

we need Langfuse instance because [`experimental_telemetry` of `streamText()` does not have attribute`cost`](https://ai-sdk.dev/docs/reference/ai-sdk-core/stream-text#experimental_telemetry.telemetry-settings).

> is it not possible to create a Custom Tracer to address this?
> https://ai-sdk.dev/docs/ai-sdk-core/telemetry#custom-tracer

yes, I think we can use our own custom tracer (not `tracerProvider.getTracer('ai')` shown in the reference) in the future.
Let me try this in another PR.

Reply:
Generation is the core of the core function in Giselle. Although you might think that telemetry could also be implemented in the same detailed way, I believe that telemetry should have a thin implementation that doesn't make its presence felt. (Alternatively, there could be a design that wraps it as a template function like in the AI SDK.) Looking forward to the implementation.

Reply:
> I believe that telemetry should have a thin implementation that doesn't make its presence felt

I agree, it's always true!
I'll refactor them into the form they should be.